components/threading/global_threads/thread_item/thread_item.tsx

```diff
@@ -80,6 +85,10 @@ function ThreadItem({
     const msgDeleted = formatMessage({id: 'post_body.deleted', defaultMessage: '(message deleted)'});
     const postAuthor = post.props?.override_username || displayName;
 
+    useEffect(() => {
+        actions.getPostThread(post.id);
+    }, []);
```

Comment:
Unfortunately, this would be called 25 times each page.
I would prefer it if we moved the api call upon opening the dot menu.

Reply:
@koox00 this was also my doubt when I was writing this in `thread_item`. 
Do you suggest to move this to `thread_menu` or `thread_pane` component?

Reply:
@koox00 I moved the logic to `thread_menu` but unfortunately it is again inside `useEffect`. This means that it will be called multiple times based on # of threads user is following (max 25 as you mentioned above)

My original idea was to call `actions.getPostThread(post.id);` when user clicks `Mark as unread` button, so it will be called only once for selected thread. . However this didnt work, (I couldnt figure it out why 😕 ). 

My guess is that I am getting `postsInThread` within `index` before I am calling `actions.getPostThread(post.id);`. Calling `actions.getPostThread(post.id);` when user clicks `Mark as unread`, `postsInThread` should be updated but currently it is not working properly. 

Please let me know if you have any suggestion. We can also jump on a quick call to discuss it

Reply:
Can we call it when menu is going to be opened?

Reply:
Funny that this same issue comes up in two different PRs for two different areas of the app at the same time https://github.com/mattermost/mattermost-webapp/pull/10755 😛 

Instead of having the component load this information, I think we should be able to make it part of the action that marks the thread as unread. Instead of just firing off the one request to mark the the thread as unread, it can check to see if the posts in the thread are loaded, load them if necessary, and then send of the request. Then, we'll only load those posts when it's absolutely necessary, and we'll only do it once per thread

Reply:
Also, if we move more of this logic into the action and out of the component like determining the value of `unreadTimestamp`, we can simplify it and potentially speed it up slightly since the component won't need that information to render. Ideally, the only props that a component receives would be actions and stuff that affects rendering, and we'd have the actions pull any other data they needed from redux directly.

Reply:
@hmhealey thanks for the guidance. I was able to implement your suggestion. It works as expected. Please feel free to suggest if you see any improvement / optimization especially for my code in `thread actions`

For `unreadTimestamp`, unfortunately I couldnt remove it because this was needed for another action (i.e. `manuallyMarkThreadAsUnread`) and not for the action I updated (i.e. `markThreadAsUnread`)